src/virt.c

```diff
@@ -578,15 +613,23 @@ lv_read (void)
                     &stats, sizeof stats) != 0)
             continue;
 
+	char *new_path=NULL;
+	if( 0==strcasecmp("true", blockdevice_format_basename) && source==blockdevice_format){ //valid only if we use source (full path to the device)
```

Comment:
*   `blockdevice_format_basename` should be of type `_Bool`
*   No Yoda Conditions please

The `if` statement then becomes:

```c
if (blockdevice_format_basename && blockdevice_format == source)
```